fs/vfs/fs_ioctl.c

```diff
@@ -39,6 +40,15 @@
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: fd_vioctl
+ ****************************************************************************/
+
+static int fd_vioctl(int fd, int req, va_list ap)
```

Comment:
The problem with sim is that it inlines this static function which messes up the variadic argument list; instead of making a copy of ap for this function call, this function consumes the first argument...